harness/determined/_core/_context.py

```diff
@@ -40,7 +40,8 @@ def __exit__(self, typ: type, value: Exception, tb: Any) -> None:
         self.distributed.close()
         # Detect some specific exceptions that are part of the user-facing API.
         if isinstance(value, det.InvalidHP):
-            self.training.report_early_exit(_core.EarlyExitReason.INVALID_HP)
+            if self.distributed._is_chief:
+                self.training.report_early_exit(_core.EarlyExitReason.INVALID_HP)
```

Comment:
the trial errors out when multiple slots post to the early_exit endpoint.  we have these lines after the post to warn about duplicate but it seems like we shouldn't post multiple times in the first place.
```
        if r.status_code == 400:
            logger.warn("early exit has already been reported for this trial, ignoring new value")
```

Reply:
what if only one non-chief worker raises an InvalidHP error?  I would argue that should be a supported use case.

Separately... does this mean that InvalidHP is broken for all dtrain right now?  because this would affect all experiments, right?

Reply:
agree that in model parallel world, we should an invalidhp exception that may be raised on a subset of ranks.  

invalidhp is broken for dtrain right now due to this line in the searcher: https://github.com/determined-ai/determined/blob/ddda7a459122c691f56b5393c649a1c9a10c93e3/master/pkg/searcher/searcher.go#L96

i'll create a ticket for us to fix in a separate PR.

Reply:
I'm posting a link to the [ticket](https://determinedai.atlassian.net/browse/DET-6792), for my own benefit when looking over review threads.

In the mean time, could you remove this if statement?  The bug is master side, and this introduces a different bug, but client-side.